### PR TITLE
Added new Social Media login options and restrict 5 socials at a single time

### PR DESCRIPTION
**Pull request by dabangarang2**
- Added support for TikTok and LinkedIn logins
- Changed Twitter logo to X
- Restricted Social login methods to 5.

fixes https://linear.app/privy/issue/PRI-1620/add-new-social-oauth-methods-to-demo

### CODE REVIEW

File: components/canvas-sidebar-auth-config.tsx [hunk 1]
```diff
@@ -137,7 +137,10 @@ export default function CanvasSidebarAuthConfig({
     setDraggedConfig(null);
   };
 
-  const hasSocials = config.loginMethods?.some((m) => ['sms', 'email'].includes(m));
+  const loginMethods = useMemo(() => config.loginMethods ?? [], [config.loginMethods]);
+  const hasSocials = loginMethods.some((m) => ['sms', 'email'].includes(m));
+  const maxLogins = loginMethods.filter((m) => !['wallet', 'sms', 'email'].includes(m)).length >= 5;
```
Comment (ankushswar1) [inline components/canvas-sidebar-auth-config.tsx:142]: nit: might rename this `reachedMaxLogins` or something a bit more descriptive

File: components/canvas-sidebar-auth-config.tsx [hunk 2]
```diff
@@ -382,7 +385,12 @@ export default function CanvasSidebarAuthConfig({
             <div className="flex flex-col gap-y-2">
               <div className="flex gap-x-4">
                 <WalletButton
```
Comment (ankushswar1) [inline components/canvas-sidebar-auth-config.tsx:387]: nit: this wasn't in scope for the ticket, but would you be able to rename `WalletButton` to `LoginMethodButton` or something similar? seems like a worthwhile refactor

File: components/canvas-sidebar-auth-config.tsx [hunk 3]
```diff
@@ -399,8 +407,8 @@ export default function CanvasSidebarAuthConfig({
                     className="shrink-0 grow-0"
                     type="checkbox"
                     name="wallet"
-                    disabled={!hasSocials}
-                    checked={config?.loginMethods?.includes('google') ?? false}
+                    disabled={disableSocials && (!loginMethods.includes('google') ?? false)}
```
Comment (ankushswar1) [inline components/canvas-sidebar-auth-config.tsx:410]: nit: it seems like we repeat this logic in a number of places:

```ts
disableSocials && (!loginMethods.includes('insert-login-method') ?? false)
```

could we refactor this into a function that accepts the `'insert-login-method'` and then reuse that function everywhere we have a `disabled` state?

File: components/canvas-sidebar-auth-config.tsx (hunk 2 above)
Comment (dabangarang2) [reply components/canvas-sidebar-auth-config.tsx:387]: sounds good will do

File: components/canvas-sidebar-auth-config.tsx (hunk 3 above)
Comment (dabangarang2) [reply components/canvas-sidebar-auth-config.tsx:410]: I was actually thinking of doing exactly that, can definitely do that refactor

**Review by ankushswar1 — APPROVED** [review]

File: components/canvas-sidebar-auth-config.tsx (hunk 1 above)
Comment (dabangarang2) [reply components/canvas-sidebar-auth-config.tsx]: completed

File: components/canvas-sidebar-auth-config.tsx (hunk 3 above)
Comment (dabangarang2) [reply components/canvas-sidebar-auth-config.tsx:411]: added this change in

**Pull request merged**